tests/test_application_summary.py

```diff
@@ -42,23 +54,41 @@ def test_serialise_application_summary():
     assert applications[1].last_edited is None
 
 
-def test_dashboard_route(flask_test_client, requests_mock):
-    requests_mock.get(
-        "http://application_store/applications?account_id=test-user",
-        text=TEST_APPLICATION_STORE_DATA,
+def test_dashboard_route(flask_test_client, mocker):
+    mocker.patch(
+        "app.default.data.get_local_data",
+        return_value=TEST_APPLICATION_STORE_DATA,
     )
+
     response = flask_test_client.get(
         "/account/test-user", follow_redirects=True
     )
     assert response.status_code == 200
-    assert b"IN PROGRESS" in response.data
+    assert b"In Progress" in response.data
+    assert b"Continue application" in response.data
     assert b"20/05/22" in response.data
 
 
-def test_dashboard_route_no_applications(flask_test_client, requests_mock):
-    requests_mock.get(
-        "http://application_store/applications?account_id=test-user",
-        text="[]",
+def test_submitted_dashboard_route_shows_no_application_link(
+    flask_test_client, mocker
+):
+    mocker.patch(
+        "app.default.data.get_local_data",
+        return_value=TEST_SUBMITTED_APPLICATION_STORE_DATA,
+    )
+    response = flask_test_client.get(
+        "/account/test-user", follow_redirects=True
+    )
+    assert response.status_code == 200
+    # there should be no link to application on the page
+    assert b"Continue application" not in response.data
+    assert b"Submitted" in response.data
+
+
+def test_dashboard_route_no_applications(flask_test_client, mocker):
+    mocker.patch(
+        "app.default.data.get_local_data",
+        return_value=[],
```

Comment:
This is a little bit odd to mock the mock response! Could we set USE_LOCAL_DATA to False and mock the real response instead?

Reply:
Fair point - I have changed the mock